### ISSUE TITLE
"Use snake_case for source file names." false positive for style linter

### ISSUE THREAD

**Issue by zedtux** [bbatsov/rubocop#2024]
I have the following offense:

```
lib/check-server-accounts.rb:1:1: C: Use snake_case for source file names.
require 'check_server_accounts/options_parser'
^
```

It looks like [the regex of this cop](https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/style/file_name.rb#L10) is not correct.

It should be changed to:

```ruby
SNAKE_CASE = /^[\/\da-z_]+$/
```

**Comment by philoserf** [comment]
See my comment in response to your comment on issue #1496

**Comment by FunTimeCoding** [comment]
+1

**Comment by brandt** [comment]
```
lib/check-server-accounts.rb
```

For whomever else stumbles upon this:

It's referring to the name of the file it's checking, not the first line of code.

I thought it was a bug too when it appeared to be complaining about the shebang.

**Comment by FunTimeCoding** [comment]
silly me, I didn't get it until now.

**Comment by lorentrogers** [comment]
This should really be more clear about what's happening. If the file name is the issue, there should be a better warning message that says so. Either that, or this needs better documentation.

**Comment by bbatsov** [comment]
@lorentrogers No argument from me. :-) I guess we should change this message.

**Comment by lorentrogers** [comment]
Awesome - I'll put together a PR soon.

**Comment by bbatsov** [comment]
👍

**Comment by lostcowboy** [comment]
So, I am new to Visual studio cold, new to the ruby extension, new to rubocop extension, new to ruby programing. I think I got everything installed right. Something complained that my file should be named main, so I changed that. then when I installed rubocop it complained that the file name doesn't match.
(The name of this source file (``) should use snake_case. (convention:Style/FileName)' at: '1,1' source: ''), So I changed it to main_1.rb . I have no ideal what the error is complaining about! But I still have the error, put it in google and found this. What is ( snake_case) file name. Duh!

**Comment by jonas054** [comment]
Looks like a problem with the editor integration. What the error message is saying is "you've given a file called '' (i.e., an empty name) to RuboCop, and that file name is not *snake case*". This means that it doesn't consist of lower case letters, digits and underscores.

Editors typically call RuboCop with the `--stdin FILE` option, but passing an empty string will not result in any inspection, so I'm not sure how you'd get that error.
